### ISSUE TITLE
[analyzer?] Show getters in debugger

### ISSUE THREAD

**Issue by asashour** [dart-lang/sdk#48825]
In debugging, only fields are shown, but not the getters. Having also getters would add a significant value I guess (specially when one is not familiar with the API used).

E.g. for:

```dart
class A {
  int x;
  int y;
  get sum => x + y;
  A(this.x, this.y);
}

void main() {
  var a = A(1, 2);
  print(a); // -< debug here
}
```

IntelliJ and Visual Studio Code show no details about `sum`:
<img src=https://user-images.githubusercontent.com/2410127/163679336-002b87b5-e6f6-401a-9b2c-68d5a62f6de4.png> and <img src=https://user-images.githubusercontent.com/2410127/163679348-268e877f-71aa-4a0b-8039-f87ece52a567.png>

Is is something that should be implemented?

**Comment by a-siva** [comment]
yes, getters are considered methods and can potentially mutate the object. Deep mutation could end up being an expensive operation that could slow down the debugging process.

**Comment by bkonyi** [comment]
@DanTup did we ever add support for displaying getters in VSCode?

**Comment by DanTup** [comment]
Yep, we display them in VS Code (you can choose between not showing, showing by default, or showing placeholders that you "expand" to trigger evaluation, to prevent the side-effects). We default to evaluating by default and despite the possibility of side-effects, I think most people prefer that.

There is an open issue at https://github.com/dart-lang/sdk/issues/46723 also about making this more efficient, because the way we currently build the list of getters across the hierarchy and then evaluate individually still feels a bit clumsy.

**Comment by bkonyi** [comment]
That's what I thought, thanks for confirming! @jwren @helin24, do we also show getter values in Intellij?

**Comment by helin24** [comment]
I don't think so; this is all I see:

<img width="238" height="207" alt="Image" src="https://github.com/user-attachments/assets/288549b5-5ef4-461f-98db-a75b9b7f59cd" />

**Comment by bkonyi** [comment]
Thanks for confirming @helin24. I'm going to move this to `area-intellij` since I think this is implemented everywhere else.
